### DISCUSSION TITLE
extracting cds or merged exons plus upstream and downstream regions

### DISCUSSION THREAD

**Discussion by sjannielefevre** [NBISweden/AGAT#418] [Q&A] [answered]
Hi

I have a genome annotation for a non-model organism, where we do not yet have the UTRs annotated, i.e. the transcripts/mrna features correspond to the cds for most genes. 

For ribosomal footprint profiling, I need to look at reads mapping just after/before the start and stop codons (to determine that the footprints map primarily to CDS). To do this, I need a transcript fasta file that have the CDS but with e.g. 100 bp added at both ends (thereby "faking" the UTRs). Just to clarify, the easiest would have been to map to the genome, but of the many different tools for riboseq qc'ing, the simplest one (that I can also get to work on my system) takes alignments to the transcriptome and not the genome.

I have looked at the AGAT options for extracting sequences (`agat_sp_extract_sequences.pl`), and there is the `--upstream` and `--downstream` options and the example given is for `-t gene` (which would include introns, so that does not work for me), but to me it is unclear whether combining it with the `-t cds` option would work as intended. 
Does anyone know or have experience with this? Or other ideas how to solve it?

My suggested command:

`agat_sp_extract_sequences.pl -g genome.gff -f genome.fasta -t cds --upstream 100 --downstream 100 -o genome.cds100bp.fasta
`

Cheers
Sjannie

**Comment by sjannielefevre** [comment]
It seems it works as intended :-) except for a few of the transcripts, where AGAT throws a warning about the extracted sequence being different than intended, likely if a transcript is closer than 100nt to the boundaries of the scaffold.

**Reply by Juke34** [reply, marked as answer]
Your command and interpretation sounds correct.
